File: swing/src/main/java/wannabe/swing/renderer/FilledSquareWithCabinetSides.java
package wannabe.swing.renderer;

import java.awt.Graphics;
import wannabe.Projected;

public class FilledSquareWithCabinetSides extends SwingRenderer {
  final Sides sides = new Sides();
  @Override public void draw(Graphics g, Projected r) {
    sides.fillSides(g, r);
    if (!r.neighborAbove) g.fillRect(r.left, r.top, r.size, r.size);
  }
}

File: swing/src/main/java/wannabe/swing/renderer/FilledSquareWithCabinetSides.kt
package wannabe.swing.renderer

import wannabe.swing.SwingProjected
import java.awt.Graphics

class FilledSquareWithCabinetSides : SwingRenderer() {
  private val sides = Sides()

  override fun draw(
    g: Graphics,
    p: SwingProjected
  ) {
    sides.fillSides(g, p)
    if (!p.neighborAbove) g.fillRect(p.left, p.top, p.size, p.size)
  }
}
